### PR TITLE
Feat/ARM-217: Code Input

### PR DESCRIPTION
**Pull request by tomRocketmakers**
## What's new?

- Renamed some functions for legibility
- Updated how value is stored. This fixes an error where a user could remove text from the middle input which would cause the rest of the text to jump around between inputs.

## Important context?
- Have also updated the "onKeyDown" function to use the correct pointers to keyboard directional arrows for switching between inputs. _However_, the behaviour of this is a bit wobbly. I think selectionStart and selectionEnd might need looking at, but definitely feels manageable.

## Ticket number(s) in JIRA (if internal)

ARM-217

[board](https://rocketmakers.atlassian.net/jira/software/projects/HM/boards/172)

## Checklist

- [ ] are your changes in Storybook?
- [ ] are any breaking changes documented in `docs/migrating_from_oldstrong.md`?
- [ ] does _everything_ have jsdoc?
- [ ] is everything exported from index.ts?
- [ ] are all new hooks added to `src/hooks/hooks.stories.mdx` or given their own docs in Storybook?
- [ ] are all new SCSS mixins added to `src/theme/mixins.stories.mdx`?
- [ ] are all new SCSS variables added to `src/theme/variables.stories.mdx`?

### CODE REVIEW

File: module/src/components/codeInput/codeInput.component.tsx [hunk 1]
```diff
@@ -133,16 +138,13 @@ export interface ICodeInputProps<TBind extends NullOrUndefined<string>>
...
-  value?: TBind;
-
   /** Fired when the code input changes */
   onChange?: (newValue: TBind) => void;
 
   /**
    * the parts of the code input
    * Can be a number representing the length of an input, I.E [1,1,1]
-   * Can be a string representing a piece of text inbetween inputs I.E. [1,1,'-',1,1]
+   * Can be a string representing a piece of text in-between inputs I.E. [1,1,'-',1,1]
```
[the remark this answers is not in the document]
Comment (tomRocketmakers) [reply module/src/components/codeInput/codeInput.component.tsx:147]: Before my time! Can get that changed though - thinking just e.g.?

File: package.json [hunk 2]
```diff
@@ -14,7 +14,8 @@
     "test": "(cd module && npm test)",
     "prepare": "husky install",
     "storybook": "(cd module && npm run storybook)",
-    "test-storybook": "(cd module && npm run test-storybook)"
+    "test-storybook": "(cd module && npm run test-storybook)",
+    "watch-scss": "(cd module && ts-node ./_build/scss/watch.ts)"
```
[the remark this answers is not in the document]
Comment (tomRocketmakers) [reply package.json:18]: My understanding was that it needs to be run when starting the project, in place of start-sass

File: module/src/components/codeInput/codeInput.component.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (NathanRHD) [reply module/src/components/codeInput/codeInput.component.tsx:147]: yeah :)

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (NathanRHD) [reply package.json:18]: Fair enough - would be good to get some docs on that in the readme!

**Pull request merged**